### PR TITLE
Clang tidy fixes

### PR DESCRIPTION
**Pull request by mcneish1**
~(work in progress)~

Look at this diff vs [this branch](https://github.com/mcneish1/GridDyn/tree/formatting_fixup) (or just https://github.com/LLNL/GridDyn/pull/28/commits/f158def93be8cccdf6d9919b77dfadb057ccce10 and onward), formatting_fixup is the result of running clang-format on everything I can compile (still missing a few things).

Before:
```
Total warnings: 1048
 * clang-analyzer-core.uninitialized.UndefReturn: 1
 * hicpp-use-auto: 1
 * performance-unnecessary-copy-initialization: 1
 * clang-diagnostic-unused-const-variable: 1
 * clang-diagnostic-unused-lambda-capture: 1
 * cert-dcl50-cpp: 1
 * readability-non-const-parameter: 1
 * clang-diagnostic-shadow: 1
 * performance-inefficient-string-concatenation: 1
 * hicpp-member-init: 1
 * clang-diagnostic-unreachable-code: 1
 * cert-err34-c: 2
 * hicpp-move-const-arg: 2
 * google-readability-casting: 2
 * llvm-namespace-comment: 2
 * clang-diagnostic-deprecated-declarations: 3
 * modernize-loop-convert: 3
 * hicpp-noexcept-move: 4
 * performance-unnecessary-value-param: 5
 * clang-diagnostic-unused-variable: 7
 * clang-analyzer-optin.cplusplus.VirtualCall: 9
 * hicpp-no-array-decay: 10
 * misc-misplaced-const: 11
 * readability-else-after-return: 11
 * clang-analyzer-core.CallAndMessage: 11
 * clang-analyzer-deadcode.DeadStores: 12
 * modernize-pass-by-value: 13
 * google-runtime-int: 14
 * hicpp-use-equals-default: 22
 * hicpp-signed-bitwise: 25
 * cppcoreguidelines-pro-type-const-cast: 28
 * cppcoreguidelines-pro-type-member-init: 31
 * clang-diagnostic-unused-parameter: 53
 * readability-named-parameter: 75
 * cppcoreguidelines-pro-bounds-array-to-pointer-decay: 83
 * readability-inconsistent-declaration-parameter-name: 91
 * readability-implicit-bool-conversion: 117
 * cppcoreguidelines-pro-type-cstyle-cast: 123
 * cppcoreguidelines-owning-memory: 126
 * hicpp-vararg: 142
```

Currently:
```
Total warnings: 802
 * clang-analyzer-core.uninitialized.UndefReturn: 1
 * google-readability-casting: 1
 * performance-unnecessary-copy-initialization: 1
 * clang-diagnostic-unused-const-variable: 1
 * cert-dcl50-cpp: 1
 * hicpp-member-init: 1
 * performance-inefficient-string-concatenation: 1
 * clang-diagnostic-unreachable-code: 1
 * cert-err34-c: 2
 * clang-diagnostic-deprecated-declarations: 3
 * modernize-loop-convert: 3
 * hicpp-noexcept-move: 4
 * performance-unnecessary-value-param: 5
 * hicpp-no-array-decay: 5
 * clang-diagnostic-unused-variable: 7
 * readability-named-parameter: 7
 * clang-analyzer-optin.cplusplus.VirtualCall: 9
 * clang-analyzer-core.CallAndMessage: 11
 * misc-misplaced-const: 11
 * clang-analyzer-deadcode.DeadStores: 12
 * modernize-pass-by-value: 13
 * google-runtime-int: 13
 * hicpp-use-equals-default: 22
 * clang-diagnostic-unused-parameter: 24
 * hicpp-signed-bitwise: 25
 * cppcoreguidelines-pro-type-const-cast: 28
 * cppcoreguidelines-pro-type-member-init: 29
 * readability-implicit-bool-conversion: 31
 * readability-inconsistent-declaration-parameter-name: 54
 * cppcoreguidelines-pro-bounds-array-to-pointer-decay: 85
 * cppcoreguidelines-pro-type-cstyle-cast: 123
 * cppcoreguidelines-owning-memory: 126
 * hicpp-vararg: 142
```

One worrying thing:
~clang-format mislabels `}  // namespace loads` in [approximating|gridLabD]Load.cpp, thinking it's supposed to be `namespace griddyn`. I'm not sure quite why, maybe something the preprocessor takes out?~

### CODE REVIEW

File: src/griddyn/loads/approximatingLoad.cpp [hunk 1]
```diff
@@ -537,7 +531,7 @@ std::vector<double> approximatingLoad::run3ApproxB ()
 }
 
 return retP;
-}
+}  // namespace loads
```
Comment (phlptp) [inline src/griddyn/loads/approximatingLoad.cpp]: this doesn't seem like a namespace closure, it is just a function termination

**Review by phlptp — COMMENTED** [review]
other than those load comments it looks pretty good, and I didn't see any issues.  We need to check if it compiles fine, since some of the headers were reordered, which can cause some issues.

File: src/griddyn/loads/approximatingLoad.cpp (hunk 1 above)
Comment (mcneish1) [reply src/griddyn/loads/approximatingLoad.cpp]: Oh oops

**Comment by mcneish1** [comment]
This one is going to be a problem for anything being developed currently (5 conflicts already), I'll revert the files I don't fix other things in if you want.

The header include order is fine, no compilation failures. There were previously a couple headers in src/griddyn that had include order issues, but I haven't checked recently or tracked that down quite yet.

**Comment by phlptp** [comment]
most of the conflicts look fairly easy to resolve.  I was wondering where the "and" and "or" came from in fmiMESubModel.  that just seemed really odd.  caused a bunch of compile errors.  I would just resolve the conflicts then we can do a merge quickly.  Hopefully this will the last major format change.

**Comment by phlptp** [comment]
why isn't this PR running through travis?

**Comment by nightlark** [comment]
It looks like the conflicts might be stopping the PR from running on Travis?

**Comment by mcneish1** [comment]
Travis didn't run on https://github.com/LLNL/GridDyn/pull/27 or the last commit of https://github.com/LLNL/GridDyn/pull/22 either.

`and` and `or` are [part of the language](https://en.cppreference.com/w/cpp/language/operator_alternative). I can do `&&` and `||` instead if some compiler doesn't like them, but they shouldn't be the original bitwise `&` and `|`

**Comment by mcneish1** [comment]
The two files that got the weird namespace markers have the same issue
```
#if 0
[...]
#else
[...]
    {
#endif
```
so after that gets fixed it shouldn't be a problem anymore.

**Comment by phlptp** [comment]
visual studio doesn't like them unless some odd compiler switches are activated.  The main reason being they are recognizable C identifiers so if they were allowed by default C code in MSVC would have a bunch of issues.  so for conformance we should stick with && and ||

**Comment by phlptp** [comment]
the #if 0 are there since there was two different ways of computing the quadratic terms,  and there is still some debate on which one is best, so for now both are in place until the issue is totally resolved.

**Comment by mcneish1** [comment]
I think it's just a matter of
```
    else
    {
        X3 = ((V2 - V1) * (Q3 - Q1) + (V1 - V3) * (Q2 - Q1)) / ((V1 - V3) * (b2 - b1) + (b1 - b3) * (V1 - V2));
        X2 = (Q2 - Q1 + b1 * X3 - b2 * X3) / (V2 - V1);
        X1 = Q1 - V1 * X2 - b1 * X3;

        retP[1] = X1;
        retP[3] = X2;
        retP[5] = X3;
    }
#endif
```
instead of 
```
    else
    {
        X3 = ((V2 - V1) * (Q3 - Q1) + (V1 - V3) * (Q2 - Q1)) / ((V1 - V3) * (b2 - b1) + (b1 - b3) * (V1 - V2));
        X2 = (Q2 - Q1 + b1 * X3 - b2 * X3) / (V2 - V1);
        X1 = Q1 - V1 * X2 - b1 * X3;
#endif

    retP[1] = X1;
    retP[3] = X2;
    retP[5] = X3;
}
```
since X1/2/3 aren't defined in the other half of the `#if`

**Comment by nightlark** [comment]
It looks there is a build for the commit yesterday, but the status isn't showing up on GitHub: https://travis-ci.org/LLNL/GridDyn/builds/405137760

**Comment by phlptp** [comment]
@mcneish1 if you can resolve these conflicts we can do a quick turnaround and get this merged.  Having all the formatting updated would be good.

**Comment by mcneish1** [comment]
Fixing appveyor, then this is good to go

**Comment by mcneish1** [comment]
https://ci.appveyor.com/project/GridDyn/griddyn/build/0.9.1.329

@phlptp or @nightlark , any idea why it's failing in cmake? That diff didn't touch the build at all

**Review by phlptp — APPROVED** [review]

**Comment by phlptp** [comment]
I am going to merge this and we can sort out the few remaining issues later

**Comment by mcneish1** [comment]
https://ci.appveyor.com/project/GridDyn/griddyn/build/0.9.1.330
No issues. Wonder what happened.

**Comment by nightlark** [comment]
It might have gone away because SuiteSparse is cached by AppVeyor.

**Pull request merged**